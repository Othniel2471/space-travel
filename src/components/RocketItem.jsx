import React from 'react';

function RocketItem() {
  return (
    <div className="container mx-auto px-3 py-3 flex flex-col md:flex-row">
      <div className="col-span-2">
        <h1 className="text-2xl font-semibold">Rockets</h1>
      </div>
      <div className="col-span-7 mt-4 md:mt-0 ml-3">
        <h3 className="capitalize text-lg font-semibold">falcon</h3>
        <p className="title">
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Hic rem vero
          eos esse beatae. Laborum ratione, saepe odio dolor cum a laboriosam
          nisi dolore velit rem, magnam, cupiditate et explicabo!
        </p>
        <button
          type="button"
          className="capitalize bg-[#017bfe] mt-2 text-white rounded py-2 px-4 hover:bg-[#0d8bf0]"
        >
          reserve rocket
        </button>
      </div>
    </div>
  );
}

export default RocketItem;
