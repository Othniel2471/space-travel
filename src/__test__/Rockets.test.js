import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import RocketItem from '../components/RocketItem';

const mockStore = configureStore([]);

describe('RocketItem', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      rockets: [
        {
          id: 1,
          name: 'Falcon 1',
          description:
            'Falcon 1 is a two-stage rocket designed and manufactured by SpaceX',
          image: 'https://images2.imgbox.com/40/e3/GypSkayF_o.png',
          reserved: false,
        },
        {
          id: 2,
          name: 'Falcon 9',
          description:
            'Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.',
          image: 'https://images2.imgbox.com/be/e7/iNqsqVYM_o.png',
          reserved: false,
        },
      ],
    });

    store.dispatch = jest.fn();

    component = render.create(
      <Provider store={store}>
        <RocketItem rocket={store.rockets} />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
