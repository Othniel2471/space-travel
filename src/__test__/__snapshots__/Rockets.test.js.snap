// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketItem should render with given state from Redux store 1`] = `
<div
  className="container mx-auto px-3 py-3 flex gap-10 flex-col md:flex-row"
>
  <div
    className="col-span-2 w-full md:w-2/12"
  >
    <img
      className="w-full h-auto object-cover"
    />
  </div>
  <div
    className="col-span-7 w-full  md:w-10/12 mt-4 md:mt-0 ml-3"
  >
    <h3
      className="capitalize text-lg font-semibold"
    />
    <p
      className="title"
    />
    <button
      className="bg-[#017bfe] mt-2 text-white rounded py-2 px-4"
      onClick={[Function]}
      type="button"
    >
      Reserve rocket
    </button>
  </div>
</div>
`;
